src/components/ModListUpdateBanner.vue

```diff
@@ -0,0 +1,78 @@
+<script lang="ts">
+import { mixins } from 'vue-class-component';
+import { Component, Watch } from 'vue-property-decorator';
+
+import UtilityMixin from './mixins/UtilityMixin.vue';
+import ManagerInformation from '../_managerinf/ManagerInformation';
+
+@Component({})
+export default class ModListUpdateBanner extends mixins(UtilityMixin) {
+    updateError = '';
+
+    get appName(): string {
+        return ManagerInformation.APP_NAME;
+    }
+
+    get isModListLoaded(): boolean {
+        return this.$store.state.tsMods.modsLastUpdated !== undefined;
+    }
+
+    get isUpdateInProgress(): boolean {
+        return this.$store.state.tsMods.isBackgroundUpdateInProgress;
+    }
+
+    @Watch('isUpdateInProgress')
+    onLoadingProgressChange(newVal: boolean, oldVal: boolean) {
+        // React to background update as well as the manual update from the banner.
+        if (!oldVal && newVal) {
+            this.updateError = '';
+        }
+    }
+
+    async updateModList() {
+        if (this.isUpdateInProgress) {
+            return;
+        }
+
+        this.$store.commit('tsMods/startBackgroundUpdate');
+        this.updateError = '';
+
+        try {
+            // Invalidate hash to force a refresh. Otherwise a scenario where
+            // the latest index hash is already present in IndexedDB but loading
+            // the package list into Vuex store has failed would cause the banner
+            // to just disappear when refreshThunderstoreModList skips the actual
+            // update but updates the timestamp of the hash.
+            await this.$store.dispatch('tsMods/updateIndexHash', 'invalidated');
+            await this.refreshThunderstoreModList();
+        } catch (e) {
+            this.updateError = `${e}`;
+        } finally {
+            this.$store.commit('tsMods/finishBackgroundUpdate');
+        }
+    }
+}
+</script>
+
+<template>
+    <div v-if="!isModListLoaded" id="mod-list-update-banner" class="margin-bottom">
+        <div class="notification is-warning margin-right">
+            <span v-if="isUpdateInProgress">
+                Updating mod list from Thunderstore...
+            </span>
+            <span v-else-if="updateError">
+                Error updating the mod list: {{ updateError }}<br />
+                {{ appName }} will keep trying to update the mod list in the background.
```

Comment:
Nit: We could generify this by using `The manager` or `We`

To counter the design a little, we could have:

```
There was an error updating the mod list. <a>Show the error</a>.
{{ appName }} will keep trying to update the mod list in the background.
```
The thinking is that the `a` tag would open a modal and we can keep the normal error preview.

Reply:
Changed to `The manager`.

I can do another PR for the error modal suggestion, but do you think it still matters what we do here if the profile import modal (in #1598 ) will show the error "inline" like here? I'd rather not use the error modal there, as that would interfere with the progress made on import modal.